specs/TransportProtocols.md

```diff
@@ -0,0 +1,176 @@
+
+This transport uses HTTP Post requests to a specific end point to encode messages. Each HTTP Request represents a single message (this achieves the Frame-based requirement), contains raw binary data (Binary-safe) and uses the query string to encode the metadata.
+
+This transport requires that a connection be established using the `[endpoint-base]/negotiate` end point.
+
+The HTTP request is made to the URL `[endpoint-base]/send` with the following query string parameters
+
+* `id` (Required) - The Connection ID of the destination connection.
+* `format` (Optional: default `text`) - Either `text` or `binary` (**case-insensitive**), indicating the type of the frame
+* `endOfMessage` (Optional: default `true`) - Indicates if this frame completes a message or if futher frames will be sent.
+
+The content of the frame is the entire Body of the HTTP request. There are no other restrictions on the Headers provided (i.e. `Content-Type`, etc.).
+
+## Server-Sent Events (Server-to-Client only)
+
+Server-Sent Events (SSE) is a protocol specified by WHATWG at [https://html.spec.whatwg.org/multipage/comms.html#server-sent-events](https://html.spec.whatwg.org/multipage/comms.html#server-sent-events). It is capable of sending data from server to client only, so it must be paired with the HTTP Post transport.
+
+The protocol is similar to Long Polling in that the client opens a request to an endpoint and leaves it open. The server transmits frames as "events" using the SSE protocol. The protocol encodes a single event as a sequence of key-value pair lines, separated by `:` and using any of `\r\n`, `\n` or `\r` as line-terminators, followed by a final blank line. Keys can be duplicated and their values are concatenated with `\n`. So the following represents two events:
+
+```
+foo: bar
+baz: boz
+baz: biz
+quz: qoz
+baz: flarg
+
+foo: boz
+
+```
+
+In the first event, the value of `baz` would be `boz\nbiz\nflarg`, due to the concatenation behavior above. Full details can be found in the spec linked above.
+
+In this transport, the client establishes an SSE connection to `[endpoint-base]/sse`, and the server responds with an HTTP response with a `Content-Type` of `text/event-stream`. Each SSE event represents a single frame from client to server. The transport uses unnamed events, which means only the `data` field is available. Thus we use the first line of the `data` field for frame metadata. The frame body starts on the **second** line of the `data` field value. The first line has the following format (Identifiers in square brackets `[]` indicate fields defined below):
+
+```
+[Type],[Fin]
+```
+
+* `[Type]` is a single UTF-8 character representing the type of the frame; `T` indicates Text, and `B` indicates Binary.
+* `[Fin]` is a single UTF-8 character indicating if the frame is the last frame of the message; `T` indicates `true`, `F` indicates `false`
```

Comment:
Yeah... `T`/`B` was to make Text/Binary clearer, but `T`/`F` for `Fin` seems weird. If we decide we don't want to do `Fin` at all, we could stick with `T`/`B` for the type field.